Translating/ModelSBML.py

```diff
@@ -0,0 +1,315 @@
+        self.docPlug = self.document.getPlugin("multi")
+        self.model = self.document.createModel()
+        self.modelPlug = self.model.getPlugin("multi")
+
+        #CREATED SPECIES TYPES REPRESENTING COMPLEXES and COMPARTMENTS
+        self.finishedCompartments = []
+        self.finishedComplexTypes = []
+
+
+    def create_species_feature_type(self, new_species_type, atomic: str, atomics: dict):
+        """Function creates species feature inside Atomic agent
+
+        :param: new_species_type -> libsbml object of new speciesType
+        :param: atomic -> <string> name of given atomic agent
+        :param: atomics -> atomic signature of the model
+        """
+        new_feature_type =new_species_type.createSpeciesFeatureType()
+        new_feature_type.setId(atomic + "_feature_type")
+        new_feature_type.setOccur(1)
+        for feature_value in atomics[atomic]:
+            new_feature_type.createPossibleSpeciesFeatureValue()\
+                .setId(atomic+"_"+feature_value)
+
+    def create_species_types_from_atomic(self, atomics: dict):
+        """Function creates speciesTypes from Atomic agents
+        :param: atomics -> atomic signature of the model in <dict>
+        """
+        for atomic in atomics:
+            new_species_type = self.modelPlug.createMultiSpeciesType()
+            new_species_type.setId("st_"+atomic)
+            if atomics[atomic] != set():
+                self.create_species_feature_type(new_species_type, atomic, atomics)
+
+
+    def create_species_types_from_structure(self, structs: dict):
+        """Function creates speciesTypes from Structure agents
+        :param: structs -> structure signature of the model in <dict>
+        """
+        for struct in structs:
+            new_species_type = self.modelPlug.createMultiSpeciesType()
+            new_species_type.setId("st_"+struct)
+            for num, subcomponent in enumerate(sorted(structs[struct]), start=0):
+                new_instance = new_species_type.createSpeciesTypeInstance()
+                new_instance.setId(subcomponent)
+                new_instance.setSpeciesType("st_"+subcomponent)
+
+    def create_species_type_from_complex(self, comp_agent : Core.Complex):
+        """All complex agents are translated to SpeciesTypes Here
+        :param: comp_agent -> <Core.Complex> -> BCSL agent to be translated to SBML-multi species type
+        """
+        new_species_type = self.modelPlug.createMultiSpeciesType()
+        name = comp_agent.to_SBML_speciesTypes_code()
+        new_species_type.setId(name)
+
+        for num, subcomponent in enumerate(sorted(comp_agent.get_agent_names()), start=0):
+            new_instance = new_species_type.createSpeciesTypeInstance()
+            new_instance.setId(subcomponent+"_"+str(num))
+            new_instance.setSpeciesType("st_"+subcomponent)
+            agent = sorted(comp_agent.agents)[num]
+            if isinstance(agent, Core.Structure.StructureAgent):
```

Comment:
Good point I need to check that

Reply:
This piece of code seems to be OK. The reason why there is only check for struct is this:

The iteration above in for loop, first layer of agents is created and put into complex agent eg. {StructA(AtomicC, AtomicD), AtomicA, StructB(AtomicE, AtomicF), AtomicB}

But because there can be only 1 layer of struct agents and atomic has only feature value, i need to verify only Struct agents and put all its content into that complex agent too (AtomicC, AtomicD) and (AtomicE, AtomicF).